docs/programmers_guide/witness_formal_spec.md

```diff
@@ -245,4 +458,114 @@ returns the first value in the specified array
 
 ### `REST(array)`
 
-returns the array w/o the first item
\ No newline at end of file
+returns the array w/o the first item
+
+### `KECCAK(bytes)`
+
+returns a keccak-256 hash of `bytes`
+
+
+## Serialization
+
+The format for serialization of everything except hashes (that we know the
+length of) is [CBOR](https://cbor.io). It is RFC-specified and concise.
+
+For hashes we use the optimization of knowing the lengths, so we just read 32
+bytes
+
+### Block Witness Format
+
+Each block witness consists of a header followed by a list of operators.
```

Comment:
Are operators and instructions the same thing?

Reply:
yeah, it is a leftover word, "operator" should not be used anywhere